Update Quick Start instructions

myhro opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

While going through the Quick Start Guide, I noticed a few incorrect suggestions. This PR updates them so that anyone following these instructions doesn't encounter the same issues.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
- `Building your project` link was broken.
- Clarify that project dependencies need to be installed.
- Running `jetpack docker install` isn't required after starting Docker containers.
- The suggested `--with-deps` argument for `jetpack build` doesn't exist.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

The only visible code change appears when trying to activate Jetpack without building it first. To resolve this:

1. Clone the repository to a temporary folder.
2. Follow the installation instructions (e.g., via Docker), but do not build the plugin using `jetpack build`.
3. Activate Jetpack at `/wp-admin/plugins.php`.

The message `Your installation of Jetpack is incomplete` will be displayed on the same page.

Initsogar commented on docs/quick-start.md:
```diff
@@ -73,7 +73,6 @@ To setup Docker:
 - Open `tools/docker/.env` and make any modifications you'd like.	
 	- It's strongly recommend you at least change `WP_ADMIN_PASSWORD` to something more secure.
 - Start the Docker container using `jetpack docker up -d` (this may take some time for the first setup)
-- Install WordPress in your Docker container using `jetpack docker install` 
```
Technically, you already got a fresh non-installed WP when starting Docker containers, but `jetpack docker install` [proceeds to install it, set up users, useful plugins and configurations](https://github.com/Automattic/jetpack/blob/ab5cd1b1ba99182c836470c3901f08a6e7d482e4/tools/docker/bin/install.sh). So I'd recommend keeping it.

```
╰─ jetpack docker install
Success: WordPress installed successfully.
Success: Updated 'blog_public' option.
Creating user: author
Creating user: editor
Creating user: subscriber
Creating user: contributor
Installing Query Monitor – The developer tools panel for WordPress (3.16.4)
Downloading installation package from https://downloads.wordpress.org/plugin/query-monitor.3.16.4.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'query-monitor'...
Plugin 'query-monitor' activated.
Success: Installed 1 of 1 plugins.
Installing WP Crontrol (1.17.0)
Downloading installation package from https://downloads.wordpress.org/plugin/wp-crontrol.1.17.0.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'wp-crontrol'...
Plugin 'wp-crontrol' activated.
Success: Installed 1 of 1 plugins.
Installing Gutenberg (19.0.0)
Downloading installation package from https://downloads.wordpress.org/plugin/gutenberg.19.0.0.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'gutenberg'...
Plugin 'gutenberg' activated.
Success: Installed 1 of 1 plugins.
Installing User Switching (1.8.0)
Downloading installation package from https://downloads.wordpress.org/plugin/user-switching.1.8.0.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'user-switching'...
Plugin 'user-switching' activated.
Success: Installed 1 of 1 plugins.
Success: Enabled 4 of 4 plugin auto-updates.
Plugin 'jetpack' activated.
Success: Activated 1 of 1 plugins.

WordPress installed. Open localhost
```

myhro replied:
Hi. Do you get this message after running `jetpack docker up -d`? All I get is:

```
$ jetpack docker install

WordPress has already been installed. Uninstall it first by running:

  jetpack docker uninstall

Command exited with status 1
```

Hence why the suggestion look redundant to me. Maybe we are using different flows? 🤔

myhro replied:
Indeed, I was trying to run it after going through the WordPress web installation. After running `jetpack docker install` right after `jetpack docker up -d`, I got the same result you mentioned:

```
$ jetpack docker install
Success: WordPress installed successfully.
Success: Updated 'blog_public' option.
Creating user: author
Creating user: editor
Creating user: subscriber
Creating user: contributor
Installing Query Monitor – The developer tools panel for WordPress (3.16.4)
Downloading installation package from https://downloads.wordpress.org/plugin/query-monitor.3.16.4.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'query-monitor'...
Plugin 'query-monitor' activated.
Success: Installed 1 of 1 plugins.
Installing WP Crontrol (1.17.0)
Downloading installation package from https://downloads.wordpress.org/plugin/wp-crontrol.1.17.0.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'wp-crontrol'...
Plugin 'wp-crontrol' activated.
Success: Installed 1 of 1 plugins.
Installing Gutenberg (19.0.0)
Downloading installation package from https://downloads.wordpress.org/plugin/gutenberg.19.0.0.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'gutenberg'...
Plugin 'gutenberg' activated.
Success: Installed 1 of 1 plugins.
Installing User Switching (1.8.0)
Downloading installation package from https://downloads.wordpress.org/plugin/user-switching.1.8.0.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'user-switching'...
Plugin 'user-switching' activated.
Success: Installed 1 of 1 plugins.
Success: Enabled 4 of 4 plugin auto-updates.
Plugin 'jetpack' activated.
Success: Activated 1 of 1 plugins.

WordPress installed. Open localhost
```

I'm updating the PR.

myhro committed:
Update Quick Start instructions

- 'Building your project' link was broken.
- Clarify that project dependencies need to be installed.
- The suggested '--with-deps' argument for 'jetpack build' doesn't
  exist.

Initsogar reviewed (approved):
Thanks for your contribution 🎉
